core/file_walker.go

```diff
@@ -4,17 +4,57 @@ import (
 	"fmt"
 	"os"
 	"path/filepath"
+	"sync"
 )
 
 type File struct {
-	Name  string
-	Size  int64
-	IsDir bool
-	Files []*File
+	Name     string
+	Size     int64
+	IsDir    bool
+	Files    []*File
+	IsMarked bool
 }
 
 type ReadDir func(dirname string) ([]os.FileInfo, error)
 
+func (f *File) ToggleIsMarked() bool {
+	f.IsMarked = !f.IsMarked
+	return f.IsMarked
+}
+
+func GetMarked(root File) []string {
+	var marked []string
+	var wg sync.WaitGroup
+	c := make(chan string, 1)
+	wg.Add(1)
+	go func() {
+		wg.Wait()
+		close(c)
+	}()
+	go getMarkedRecursive("", &root, c, &wg)
```

Comment:
I really like your parallel code. And I've learnt that you can use `for .. range` on a channel just from reading your PR. I wonder if parallelism here increases the complexity of GetMarked and will make it harder to test? Have you tried the function running in one thread?